upload_log_files.py

```diff
@@ -0,0 +1,77 @@
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file-path",
+                        help="Path to the pipeline configuration json file")
+    parser.add_argument("memory_profile_dir_path", metavar="memory-profile-dir-path",
+                        help="Path to the memory profile log directory with file to upload")
+    parser.add_argument("data_archive_dir_path", metavar="data-archive-dir-path",
+                        help="Path to the data archive directory with file to upload")
+
+    args = parser.parse_args()
+
+    user = args.user
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    memory_profile_dir_path = args.memory_profile_dir_path
+    data_archive_dir_path = args.data_archive_dir_path
+
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+    Logger.set_project_name(pipeline_configuration.pipeline_name)
+    log.debug(f"Pipeline name is {pipeline_configuration.pipeline_name}")
+
+    latest_memory_profile_file_path = fetch_latest_modified_file_path(memory_profile_dir_path)
+    latest_data_archive_file_path = fetch_latest_modified_file_path(data_archive_dir_path)
+    try:
+        memory_profile_upload_location = f"{pipeline_configuration.memory_profile_upload_url_prefix}{os.path.basename(latest_memory_profile_file_path)}"
+        log.info(f"Uploading the most recent memory profile from {latest_memory_profile_file_path} to {memory_profile_upload_location}...")
+        with open(latest_memory_profile_file_path, "rb") as f:
+            google_cloud_utils.upload_file_to_blob(
+                google_cloud_credentials_file_path, memory_profile_upload_location, f
+            )
+
+        data_archive_upload_location = f"{pipeline_configuration.data_archive_upload_url_prefix}{os.path.basename(latest_data_archive_file_path)}"
+        log.info(f"Uploading the most recent data archive from {latest_data_archive_file_path} to {data_archive_upload_location}...")
+        with open(latest_data_archive_file_path, "rb") as f:
+            google_cloud_utils.upload_file_to_blob(
+                google_cloud_credentials_file_path, data_archive_upload_location, f
+            )
+        log.warning(f"Deleting old memory profile & data archives files from local disk")
+        delete_old_log_files(memory_profile_dir_path, latest_memory_profile_file_path)
+        delete_old_log_files(data_archive_dir_path, latest_data_archive_file_path)
+
+    except BaseException as ex:
```

Comment:
This is quite extreme. Two questions:
1. Is it safe to catch BaseException (not seen it before in Python - but I know that in Java it would be very unusual to catch the base throwable class because it can include errors that you wouldn't be reasonably expected to recover from, or which it may not be possible to recover from)?
2. Do you actually need to sequence it like this? You could remove all the other logs first, then upload.

Reply:
btw if you want to run something 'unconditionally', use:
```python
try:
    # Code that might fail
finally:
    # Code that should be run irrespective of the outcome of the try 
```

Reply:
updated to:
- catch know errors (ConnectionError, socket.timeout, Timeout)
- delete old files before uploading